curl_cffi/requests/__init__.py

```diff
@@ -42,7 +42,7 @@ def request(
     files: Optional[Dict] = None,
     auth: Optional[Tuple[str, str]] = None,
     timeout: Union[float, Tuple[float, float]] = 30,
-    allow_redirects: bool = True,
+    allow_redirects: bool = False,
```

Comment:
It was kept as `True` to be consistent with `requests`.

Reply:
httpx [reason](https://www.python-httpx.org/compatibility/#redirects):
> Unlike requests, HTTPX does not follow redirects by default.
> 
> We differ in behaviour here [because auto-redirects can easily mask unnecessary network calls being made](https://github.com/encode/httpx/discussions/1785).

Reply:
Also, I noticed curl-cffi uses infinity redirects (?)
Even in `requests` the default `max_redirects` is 30:
> ### max_redirects
> Maximum number of redirects allowed. If the request exceeds this limit, a [TooManyRedirects](https://requests.readthedocs.io/en/latest/api/#requests.TooManyRedirects) exception is raised. This defaults to requests.models.DEFAULT_REDIRECT_LIMIT, which is 30.

Reply:
The ship has sailed, I don't think we should make a break change. However, we can reconsider this when planning 1.0

Reply:
Besides, it's set to `True` in aiohttp. For anyone coming from that, it means less suprises.

Reply:
Yeah, max_redirects prabably should be changed.

Reply:
IIUC next release is `0.6.0`?
IIRC, According to SEMVER specification, it's fine to include breaking changes between minor releases in 0.x

btw, if you think deprecation process must apply here, I'm also OK with that ;)

Reply:
> For anyone using the AsyncSession and coming from that, it means less suprises

I'm usually use httpx in python, at first when I used curl-cffi, I suprised that I should use `allow_redirects=False` for every `s.get|post()`. because redirects can cause unwanted behaviors.

btw, allowing to configure it in AsyncSession class can help me on this. I'm fine to use `allow_redirects=False` one time on session initializing.